api/provider/aws/task_create.go

```diff
@@ -1,7 +1,90 @@
 package aws
 
-import "github.com/quintilesims/layer0/common/models"
+import (
+	"fmt"
 
-func (t *TaskProvider) Create(req models.CreateTaskRequest) (*models.Task, error) {
-	return nil, nil
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (t *TaskProvider) Create(req models.CreateTaskRequest) (string, error) {
+	taskID := generateEntityID(req.TaskName)
+	fqEnvironmentID := addLayer0Prefix(t.Config.Instance(), req.EnvironmentID)
+
+	deployName, deployVersion, err := lookupDeployNameAndVersion(t.TagStore, req.DeployID)
```

Comment:
Possibly a controversial opinion, but i'd rather `lookupDeployNameAndVersion` returned a `models.Deploy`

Reply:
we made this function because we need to do operations on the deploy's name and version (for aws calls); returning `models.Deploy` would be very odd here.